### ISSUE TITLE
RegularPatterns suggestions

### ISSUE THREAD

**Issue by ChShersh** [jaspervdj/stylish-haskell#158]
When I'm using `stylish-haskell` for next file: http://lpaste.net/3468234723434692608
I got
```
$ stylish-haskell bench/Main.hs
Language.Haskell.Stylish.Parse.parseModule: could not parse bench/Main.hs: ParseFailed 
(SrcLoc "<unknown>.hs" 25 7) "RegularPatterns language extension is not enabled. 
Please add {-# LANGUAGE RegularPatterns #-} pragma at the top of your module."
```

I know it relates to bug in `haskell-src-exts`: https://github.com/haskell-suite/haskell-src-exts/issues/305
But maybe some workarounds can be made on `stylish-haskell` side. Obviously, if I comment next line:

```haskell
24:      [ bench "fmt" $ simpleBench @Text (\(a,b) -> "hello"#|a|#"world"#|b|#"")
```
everything works fine, problem is gone. But commenting such code is not a solution for me :)

Though after that I'm experiencing another strange behavior. Line

```haskell
import           Fmt            ((#|), (|#))
```

replaces with

```haskell
import           Fmt             (( #| ), (|#))
```

I'm using `stylish-haskell 0.7.1.0`.

**Comment by neongreen** [comment]
I think the latter thing is due to the fact that without spaces `(#|)` might trip the GHC parser if `-XMagicHash` is enabled (because `(#` is parsed as the start of an unboxed tuple).

**Comment by neongreen** [comment]
The original example has been fixed in #160 (not released yet). I will file the `( #| )` thing as a separate issue.

**Comment by neongreen** [comment]
https://github.com/jaspervdj/stylish-haskell/issues/161
